packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,196 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextField from '../TextField/TextField';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * Determine if a React component is a TextField
+ * @param {React.Node} child - a React component
+ * @return {Boolean} Is this a TextField component?
+ */
+function isTextField(child) {
+  return child != null && child.type === TextField;
+}
+
+/**
+ * The `Autocomplete` component is a parent component that adds autocomplete functionality to a `TextField` component.
+ */
+export class Autocomplete extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+    this.labelId = uniqueId('constrained-list-header_');
+  }
+
+  filterItems(
+    items,
+    inputValue,
+    getInputProps,
+    getItemProps,
+    highlightedIndex
+  ) {
+    if (this.props.loading) {
+      return (
+        <li className="ds-c-autocomplete__list-item--message">Loading...</li>
```

Comment:
We'll need to localize this string, so will need a prop (`PropTypes.node`) for defining it (but this could be it's default).

Reply:
Done. Added a `PropType.node` and default string of `Loading...`.